src/hotspot/share/gc/z/zUtils.cpp

```diff
@@ -38,5 +36,7 @@ const char* ZUtils::thread_name() {
 }
 
 void ZUtils::fill(uintptr_t* addr, size_t count, uintptr_t value) {
-  std::fill_n(addr, count, value);
+  for (uintptr_t* end = addr + count; addr < end; ++addr) {
+    *addr = value;
+  }
```

Comment:
I tend to avoid changing values of the input arguments, so I would like to see that changed. Unless there's a problem with the below code I would like to see this changed to this:
```
  for (uintptr_t* current = addr; current < addr + count; ++current) {
    *current = value;
  }
```

Or maybe even:
```
  for (size_t i = 0; i < count; ++count) {
    *(addr + i) = value;
  }
```

Reply:
Okay. I went with something like the 2nd suggestion, though with array syntax
rather than explict pointer arithmetic.